internal/sim/sim.go

```diff
@@ -189,10 +190,20 @@ type op struct {
 	components []reflect.Type // Op.Func component argument types
 }
 
+// fate dictates if and how a call should fail.
+type fate int
+
+const (
+	dontFail           fate = iota // dont fail a call
+	failBeforeDelivery             // fail a call before it is delivered
```

Comment:
Can you clarify here what "delivered" means.

Reply:
Done. Let me know if you think I should elaborate further :)